src/openms/include/OpenMS/KERNEL/SpectrumHelper.h

```diff
@@ -66,6 +70,34 @@ namespace OpenMS
     return it;
   }
 
+  template <typename PeakContainerT>
+  void slicePeakContainer(PeakContainerT& p, const double pos_start, const double pos_end)
+  {
+    typename PeakContainerT::iterator it_start = p.PosBegin(pos_start);
+    typename PeakContainerT::iterator it_end = p.PosEnd(pos_end);
+    p.erase(it_end, p.end());
+    p.erase(p.begin(), it_start);
+    // Note: only raw peak data is erased
+  }
+
+  template <typename PeakContainerT>
+  void rebaseIntensities(PeakContainerT& p)
+  {
+    typename PeakContainerT::iterator it = std::min_element(p.begin(), p.end(),
+      [](typename PeakContainerT::PeakType& a, typename PeakContainerT::PeakType& b)
+      {
+        return a.getIntensity() < b.getIntensity();
+      });
+
+    if (it == p.end() || it->getIntensity() >= 0) return;
+
+    const double rebase = std::fabs(it->getIntensity());
+    for (typename PeakContainerT::PeakType& peak : p)
+    {
+      peak.setIntensity(peak.getIntensity() + rebase);
```

Comment:
What if the lowest intensity is a positive value?  Wouldn't that just increase the peak height further for all points instead of move the baseline to zero?

Reply:
If the minimum intensity is a positive value, the method simply returns: https://github.com/biosustain/OpenMS/pull/159/files#diff-ede4000867d6e764797cee68ae0340a5R92
Is this the desired behaviour or would you like something different?